lib/src/main/java/nl/knaw/dans/lib/dataverse/DatasetApi.java

```diff
@@ -93,13 +97,59 @@ protected DatasetApi(HttpClientWrapper httpClientWrapper, String id, boolean isP
         return httpClientWrapper.postJsonString(path, "", parameters, new HashMap<>(), DatasetPublicationResult.class);
     }
 
+    /**
+     * Edits the current draft's metadata, adding the fields that do not exist yet. If `replace` is set to `false`, all specified
+     * fields must be either currently empty or allow multiple values. Replaces existing data.
+     *
+     * @see [[https://guides.dataverse.org/en/latest/api/native-api.html#edit-dataset-metadata]]
+     * @param s JSON document containing the edits to perform
+     * @return
+     */
+    public DataverseResponse<DatasetVersion> editMetadata(String s) throws IOException, DataverseException {
+        return editMetadata(s, true);
+    }
+
+    /**
+     * Edits the current draft's metadata, adding the fields that do not exist yet. If `replace` is set to `false`, all specified
+     * fields must be either currently empty or allow multiple values.
+     *
+     * @see [[https://guides.dataverse.org/en/latest/api/native-api.html#edit-dataset-metadata]]
+     * @param s       JSON document containing the edits to perform
+     * @param replace whether to replace existing values
+     * @return
+     */
+    public DataverseResponse<DatasetVersion> editMetadata(String s, Boolean replace) throws IOException, DataverseException {
+        log.trace("ENTER");
+        HashMap<String, List<String>> queryParams = new HashMap<>();
+        if (replace)
+            queryParams.put("replace", Collections.singletonList("true"));
+        return putToTarget("editMetadata", s, queryParams, DatasetVersion.class); // Sic! any value for "replace" is interpreted by Dataverse as "true", even "replace=false"
```

Comment:
You copied the comment from the scala lib, but put it in a peculiar place. Line 126 doesn't reference "replace"!

Reply:
done